server/src/test/java/org/elasticsearch/search/query/QueryPhaseTimeoutTests.java

```diff
@@ -125,27 +125,35 @@ private static ContextIndexSearcher newContextSearcher(IndexReader reader) throw
         );
     }
 
-    public void testScorerTimeoutTerms() throws IOException {
+    /**
+     * Test that a timeout is appropriately handled when the directory reader raises it while loading terms enum
+     * as the scorer supplier is requested.
+     */
+    public void testScorerSupplierTimeoutTerms() throws IOException {
         assumeTrue("Test requires more than one segment", reader.leaves().size() > 1);
         int size = randomBoolean() ? 0 : randomIntBetween(100, 500);
-        scorerTimeoutTest(size, context -> {
+        scorerSupplierTimeoutTest(size, context -> {
```

Comment:
I've been reading TimeoutQuery and MatchAllWeight defined in this test class but I'm still not sure how the function passed in triggers the timeout. At least that's what I'm supposing its doing given its name. Maybe its possible to explain this for future readers somehow?

Reply:
I thought the new javadocs would help explaining this. 

> Test that a timeout is appropriately handled when the (exitable) directory reader raises it while loading terms enum as the scorer is retrieved from the scorer supplier.

The idea of these tests is that a timeout can be raised from different places: either because we wrap the bulk scorer with a cancellable one, or via the exitable directory reader that wraps the original reader. Because a timeout can be raised from different places, it's easy to miss and to forget handling the corresponding exception. The reason why the timeout is thrown is because we flip shouldTimeout in the search context. I am adding a bit more comments.